nodejsalert-ui/Jenkinsfile

```diff
@@ -75,15 +150,20 @@ def promoteServiceSetup(openShiftHost, openShiftToken, svcName, registry, imageN
       echo "skip dc/svc/route creation related exception, the resource may already exist. " + e.getMessage();
     }
 }
-def promoteService (imageNamespace, projName, svcName, sourceTag, destinationTag) {
-    openshiftTag(namespace: imageNamespace,
-                  srcStream: svcName,
-                  srcTag: sourceTag,
-                  destStream: svcName,
-                  destTag: destinationTag)
-
+def tagImage(imageNamespace, projName, svcName, sourceTag, destinationTag) {
+    script {
+        openshift.withCluster() {
+            openshift.withProject( imageNamespace ) {
+                echo "tagging the build for ${MODULE_NAME} ${sourceTag} to ${destinationTag} in ${imageNamespace} "
+                openshift.tag("${MODULE_NAME}:${sourceTag}", "${MODULE_NAME}:${destinationTag}")
+            }
+        }
+    }//script
+}
+def promoteService (imageNamespace, projName, dcName) {
+    echo "Deploying service ${projName} to ${dcName}"
     openshiftDeploy(namespace: projName,
-  			     deploymentConfig: svcName,
+  			     deploymentConfig: dcName,
 			     waitTime: '300000')
 
 }
```

Comment:
what about other services(user service, alert service and gateway service ), blue green pipelines deployment strategy ?

Reply:
Consider this is a demonstration purpose and one module should be enough for user to get the concept.